WordPress/gutenberg
# Media & Text Block: Fix deprecation with `isStackOnMobile` default value changed

Fixes #29109

## What?

This PR fixes a broken deprecation related to the default value of `isStackedOnMoobile` changing from `false` to `true`.

## Why?

In #14364, the default value of the `isStackedOnMobile` attribute has been updated from `false` to `true`. At the same time, deprecations were also made and [the based attributes were defined as an object named `baseAttributes`](https://github.com/WordPress/gutenberg/pull/14364/files#diff-1f1d3f8d14ce5f982caa2de0c762395d1a87a02e8db1ebf11fcbe4b8e2d07331R22-R64). The `baseAttributes` variable is the base definition for the future deprecation, so I think this object must not be updated. 

However, in #26041 the default value of `isStackOnMobile` for `baseAttributes` was changed to `true`. This means that blocks from `v0` to `v2`, where the default value of `isStackOnMobile` is `false`, are no longer migrated.

## How?

I have reverted the definition of `baseAttributes` back to the original definition and renamed it to `v0Atributes` to prevent the same problem from occurring in the future. At the same time, I added a property to override `baseSAttributes` in v3 and later, where the default value of `isStackedOnMobile` is expected to be `true`.

## Testing Instructions

In the code editor, insert a version of the markup in which the default value of `isStackedOnMobile` is `false`:

```html

<div class="wp-block-media-text alignwide"><figure class="wp-block-media-text__media"></figure><div class="wp-block-media-text__content">
<p class="has-large-font-size"></p>
</div></div>

```

Verify that the markup updates as follows when the focus is removed from the code editor:

```html

<div class="wp-block-media-text alignwide"><figure class="wp-block-media-text__media"></figure><div class="wp-block-media-text__content">
<p class="has-large-font-size"></p>
</div></div>

```
